Documentation/Architecture/events.md

```diff
@@ -120,16 +120,32 @@ precision on using [primitive types](#Primitive Types).
 In mathematics you find the concept of [idempotence](https://en.wikipedia.org/wiki/Idempotence);
 you should be able to apply operations multiple times without it changing the result. In an event based
 system with an [event store](./event_store.md) you'll find this concept mentioned as well.
-With an [event store](./event_store.md) you enable the possibility to replay events at any given point
-in time. It is therefor important that events have enough information on them to enable this in
-conjunction with the [event processors](./event_processor.md). This does not mean that you should be
-keeping full entities on the event - still follow the rule of [complexity](#Complexity) as mentioned and
-remember to model the actual state change. Be aware of this and model accordingly.
+With an [event store](./event_store.md) you enable the possibility to replay events. You do this
+for instance if you want to get rid of all your read models, for instance if you're changing storage mechanism.
+To be able to get to the same result, you must have enough information on your events that the
+[event processors](./event_processor.md) make use of. This means that your event processor should not
+be looking up data somewhere else to get to the result. Looking something up to be able to produce the
+result creates a dependency making it impossible to guarantee the result.
 
-In addition to events being modelled correctly and the processor doing it right, the system
-must guarantee ordering of events when replaying and also that your [event processor](./event_processor.md)
-can assume that events are being delivered once and once only. This is built into the underlying
-system being used. You as a developer do not have to worry about this.
+This does not mean that you should be keeping full entities on the event - still follow the rule of
+[complexity](#Complexity) as mentioned and remember to model the actual state change.
+Be aware of this and model accordingly.
+
+Important to keep in mind that the snapshot of a state is a series of events, these together form
+the snapshot. This should reduce what you keep on them but as a whole they help guarantee the
```

Comment:
Seems like somethings missing here

Reply:
LOL..   Me fix..   Probably a good idea to finish off sentences..